src/modules/personenkontext/persistence/dbiam-personenkontext.repo.ts

```diff
@@ -276,4 +276,39 @@ export class DBiamPersonenkontextRepo {
     public async isRolleAlreadyAssigned(id: RolleID): Promise<boolean> {
         return (await this.findByRolle(id)).length > 0;
     }
+
+    public async getPersonenKontexteWithExceedingBefristung(): Promise<Record<string, Personenkontext<true>[]>> {
+        const filters: QBFilterQuery<PersonenkontextEntity> = {
+            personId: {
+                $in: this.em
+                    .createQueryBuilder(PersonenkontextEntity, 'pk')
+                    .select('pk.person_id')
+                    .where({ befristung: { $lt: new Date() } })
+                    .getKnexQuery(),
+            },
+        };
+
+        const personenKontexteEntities: PersonenkontextEntity[] = await this.em.find(PersonenkontextEntity, filters);
+        const personenKontexte: Personenkontext<true>[] = personenKontexteEntities.map((pk: PersonenkontextEntity) =>
+            mapEntityToAggregate(pk, this.personenkontextFactory),
+        );
+
+        // Grouping the entities by personId
+        const groupedByPersonId: Record<string, Personenkontext<true>[]> = personenKontexte.reduce(
+            (groups: Record<string, Personenkontext<true>[]>, personKontext: Personenkontext<true>) => {
+                const personId: string = personKontext.personId;
+
+                if (!groups[personId]) {
+                    groups[personId] = [];
+                }
+
+                // Use non-null assertion here
+                groups[personId]!.push(personKontext); // Now TypeScript knows this can't be undefined
+                return groups;
+            },
+            {} as Record<string, Personenkontext<true>[]>,
+        );
```

Comment:
Please use a Map instead of an object
i.e.
```typescript
const groupedByPerson: Map<PersonID, Personenkontext<true>[]> = new Map();
for (const kontext of personenKontexte) {
    const group: Personenkontext<true>[] = groupedByPerson.get(kontext.personId) ?? [];
    if (group.length === 0) {
        groupedByPerson.set(kontext.personId, group);
    }
    group.push(kontext);
}
```

Reply:
Changed to Map